test/suites/integration/validation_test.go

```diff
@@ -253,6 +253,12 @@ var _ = Describe("Validation", func() {
 			Expect(env.Client.Update(env.Context, nodeClass)).ToNot(Succeed())
 		})
 		It("should fail to switch between a managed and unmanaged instance profile", func() {
+			// Skipping this test for private cluster because there is no VPC private endpoint for the IAM API. As a result,
+			// you cannot use the default spec.role field in your EC2NodeClass. Instead, you need to provision and manage an
+			// instance profile manually and then specify Karpenter to use this instance profile through the spec.instanceProfile field.
+			if env.PrivateCluster {
+				Skip("skipping Extended Resources test for private cluster")
```

Comment:
```suggestion
				Skip("Skipping Unmanaged instance profile test for private cluster")
```

Can we make sure that skip statement properly models why we are skipping it?